In [1]:
import pandas as pd

In [2]:
RAG = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 02.06.2025 dataset for evaluation\\psychiatry_train_dataset_groq_mistral-saba-24b_answered_rag_answer_correctness_evaluated.csv"

In [3]:
VANILLA = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 02.06.2025 dataset for evaluation\\psychiatry_train_dataset_groq_mistral-saba-24b_answered_vanilla_answer_correctness_evaluated.csv"

In [4]:
# Read the files
rag_df = pd.read_csv(RAG)  # Replace with actual filename
vanilla_df = pd.read_csv(VANILLA)  # Replace with actual filename

# Calculate means
rag_mean = rag_df['Mean Answer Correctness for RAG'].mean()
vanilla_mean = vanilla_df['Mean Answer Correctness for vanilla'].mean()

print(f"RAG Mean Answer Correctness: {rag_mean}")
print(f"Vanilla Mean Answer Correctness: {vanilla_mean}")

RAG Mean Answer Correctness: 0.4123253574398246
Vanilla Mean Answer Correctness: 0.4220597875819323


The very first results were:  
RAG Mean Answer Correctness: 0.4123253574398246
Vanilla Mean Answer Correctness: 0.4220597875819323  

Addition of Naive RAG deteriorates the answer quality. 
